src/InfoMCStructHelper.cc

```diff
@@ -174,6 +160,19 @@ namespace mu2e {
         break; // this particle doesn't have a parent
       }
     }
+
+    // Now add all the primary particles
+    SimInfo sim_info;
+    for(auto const& spp : primary.primarySimParticles()){
+      // check whether we already put this primary in
+      fillSimInfo(spp, sim_info);
+      sim_info.trkrel = MCRelationship(spp, trkprimaryptr);
+      sim_info.prirel = MCRelationship(spp, spp);
+      if (sim_info.trkrel != MCRelationship::same) {
```

Comment:
I think you want 'sim_info.trkrel == MCRelationship::none.  The primary could match a parent or grandparent and so already be in the siminfo vector.  A safter test would be to explicitly loop over all the siminfo entries testing !=same.

In case the particle is already in the siminfo vector, I think you want to update the prirel value.  Or does that already get filled in fillSimInfo?

Reply:
Thanks, Dave. The prirel and trkrel values get filled in this function ```fillAllSimInfos``` instead of ```fillSimInfo``` but ```prirel``` and ```trkrel``` are set by now so I should be able to look through the SimInfos already in the vector and to check them

Reply:
The safer test has been included in commit [478e135](https://github.com/Mu2e/TrkAna/pull/101/commits/478e1354f95fcf713ce5644b69064665e8bab478)